### PR TITLE
CMR-6164

### PR DESCRIPTION
**Pull request by daniel-zamora**

### CODE REVIEW

**Comment by daniel-zamora** [comment]
PR is a little pre mature, there is currently no tests for this job.

File: ingest-app/src/cmr/ingest/services/jobs.clj [hunk 1]
```diff
@@ -263,37 +264,49 @@
...
+(defn- filter-gran-refs-by-subscriber-id
+  "Takes a list of granule references and a subscriber id and removes any granule that the user does
+   not have read access to."
+  [context gran-refs subscriber-id]
+  (let [concept-ids (map :concept-id gran-refs)
+        permissions (json/parse-string
+                     (access-control/get-permissions context {:user_id subscriber-id
+                                                              :concept_id (map :concept-id gran-refs)}))]
+    (filter (fn [granule-reference]
+              (some #{"read"} (get permissions (:concept-id granule-reference))))
```
Comment (ygliuvt) [inline ingest-app/src/cmr/ingest/services/jobs.clj:276]: What about granules that are public?

**Review by ygliuvt — CHANGES_REQUESTED** [review]
Need to consider public granules when filtering by subscriber id.

**Comment by ygliuvt** [comment]
yes, we should add a test for this.

> PR is a little pre mature, there is currently no tests for this job.

**Review by jceaser — APPROVED** [review]
thank you for fixing my code alignment!

File: ingest-app/src/cmr/ingest/services/jobs.clj [hunk 2]
```diff
@@ -263,42 +264,61 @@
...
```
Comment (ygliuvt) [inline ingest-app/src/cmr/ingest/services/jobs.clj:276]: I want to confirm that this check should include any granules that are accessible by registered user. Are you sure this check will include those?

Comment (daniel-zamora) [reply ingest-app/src/cmr/ingest/services/jobs.clj:276]: yes.

Comment (ygliuvt) [reply ingest-app/src/cmr/ingest/services/jobs.clj:276]: cool. Make sure the test has a granule covers this case (i.e. accessible by registered user).

**Review by ygliuvt — APPROVED** [review]

**Pull request merged**